### ISSUE TITLE
Destructed assignment on first line of for-in loop emits incorrectly

### ISSUE THREAD

**Issue by thebanjomatic** [decaffeinate/decaffeinate#231]
When a destructured assignment is the first statement within a for-in loop, the ES6 code that is emitted is inconsistent with destructured assignments elsewhere:

```coffee
for entry in someArray
  {x, y} = getPoint(entry)
  console.log x + ', ' + y;
```

produces:

```javascript
for (let i = 0; i < someArray.length; i++) {
  let x, y;
  let entry = someArray[i]
  ({x, y} = getPoint(entry));
  console.log(x + ', ' + y);
}
```

Expected result:

```javascript
for (let i = 0; i < someArray.length; i++) {
  let entry = someArray[i];
  let {x, y} = getPoint(entry);
  console.log(x + ', ' + y);
}

**Comment by eventualbuddha** [comment]
Yep, it would better be represented as you say. You're not seeing a behavior difference, though, right?

**Comment by thebanjomatic** [comment]
Actually, the generated code produces an error. The lack of semi-colon after someArray[i] and the parenthesis around the destructured assignment wind up getting parsed as a function invocation:
```javascript
let entry = someArray[i]({x, y} = getPoint(entry));
```
With the ES6 `let` keyword, this produces the error `ReferenceError: entry is not defined`, because we are using the LHS identifier in the RHS of the initialization statement.

Naively switching that to `var`, it produces the runtime error `TypeError: someArray[i] is not a function`.

**Comment by eventualbuddha** [comment]
Ah, I see. Thanks for the explanation.

**Comment by eventualbuddha** [comment]
Fixed in v2.11.2 ([repl](http://decaffeinate.github.io/decaffeinate/repl/#?evaluate=true&code=for%20entry%20in%20someArray%0A%20%20%7Bx%2C%20y%7D%20%3D%20getPoint(entry)%0A%20%20console.log%20x%20%2B%20'%2C%20'%20%2B%20y%3B)).
